### ISSUE TITLE
RF04-RF06: T-SQL quoted identifier and special character detection issues

### ISSUE THREAD

**Issue by Fank** [quarylabs/sqruff#1711]
## Problem

T-SQL quoted identifier keyword detection and special character handling is not working correctly in rules RF04-RF06, affecting identifier validation and quoting policies.

### 1. RF04: Quoted Identifier Keyword Detection
```sql
-- Issue: T-SQL quoted identifier keywords not properly detected
SELECT [ORDER] FROM table1  -- Should pass validation
SELECT [SELECT] AS alias    -- Should pass validation  
```

### 2. RF05: Special Character Detection  
```sql
-- Issue: T-SQL special characters in identifiers not detected
SELECT [my-column] FROM table1     -- Should require quotes
SELECT [column with spaces] FROM table1  -- Should require quotes
```

### 3. RF06: Quoted Column Detection
```sql
-- Issue: T-SQL quoted columns not properly identified
SELECT [quoted_column] FROM table1  -- Should be detected as quoted
```

## Root Cause

1. **T-SQL bracket syntax `[name]`** not recognized by identifier detection logic
2. **Keyword validation** assumes standard double quotes, not T-SQL brackets
3. **Special character patterns** don't account for T-SQL identifier rules  
4. **Quote detection logic** hardcoded for SQL standard syntax

### Failing Tests
- `test_pass_valid_quoted_identifier_not_alias` (RF04)
- `test_pass_valid_quoted_identifier_table_alias_column_alias_config` (RF04)
- `test_fail_special_chars_unquoted_policy_aliases` (RF05)
- `test_fail_special_chars_unquoted_policy_column_aliases` (RF05)
- `test_fail_quoted_column_tsql` (RF06)

### Current Behavior
All tests ignored with message: "T-SQL quoted identifier/special character detection not working correctly"

### Location
- **Rules**: RF04, RF05, RF06 in `/crates/lib/src/rules/references/`
- **System**: Identifier analysis and quote detection logic

### RF04: Quoted Identifier Keywords
```sql
-- Should PASS (quoted keywords are allowed)
SELECT [ORDER] FROM table1
SELECT [SELECT] AS [GROUP]

-- Should FAIL (unquoted keywords not allowed)  
SELECT ORDER FROM table1
SELECT SELECT AS GROUP
```

### RF05: Special Character Policies
```sql
-- Should PASS with quoted policy
SELECT [my-column], [column with spaces] FROM table1

-- Should FAIL with unquoted policy
SELECT my-column, column with spaces FROM table1
```

### RF06: Quote Requirement Detection
```sql
-- Should REQUIRE quotes
SELECT [quoted-column] FROM table1  -- Has special chars

-- Should NOT require quotes  
SELECT simple_column FROM table1    -- No special chars
```

### 1. T-SQL Identifier Recognition
- Update identifier detection to recognize `[name]` as quoted
- Add T-SQL bracket syntax to quote detection patterns
- Handle both `[brackets]` and `"standard"` quote styles

### 2. Keyword Validation Enhancement
- Extend keyword detection for T-SQL reserved words
- Add T-SQL-specific keyword lists
- Support bracket-quoted keyword validation

### 3. Special Character Detection  
- Add T-SQL-specific special character patterns
- Support hyphens, spaces in bracketed identifiers
- Update validation rules for T-SQL identifier syntax

### 4. Quote Policy Implementation
- Implement T-SQL bracket quote policies
- Add configuration for bracket vs double-quote preferences
- Support mixed quoting styles in same query

## Impact

- **Priority**: Medium - affects T-SQL identifier validation
- **Scope**: Requires identifier analysis system updates
- **Use case**: Essential for T-SQL code style enforcement
- **Compatibility**: Maintain support for standard SQL quoting

## Implementation Notes

T-SQL identifier rules:
- `[name]` - always quoted, allows special chars/keywords
- `"name"` - quoted in ANSI mode
- `name` - unquoted, must follow identifier rules
- Special chars: hyphens, spaces, Unicode in brackets

## Labels
- `enhancement`
- `dialect:tsql`
- `rule:RF04`
- `rule:RF05` 
- `rule:RF06`
- `identifier-analysis`

**Comment by benfdking** [comment]
Enabling the RF04 rules https://github.com/quarylabs/sqruff/pull/1722

**Comment by benfdking** [comment]
Enabling RF06 https://github.com/quarylabs/sqruff/pull/1723
